cloudsql/postgresql/server.js

```diff
@@ -0,0 +1,241 @@
+  // Additional connections will be established to meet this value unless the pool is full.
+  knex.client.pool.min = 5;
+  // [END cloud_sql_limit_connections]
+  // [START cloud_sql_connection_timeout]
+  // 'acquireTimeoutMillis' is the maximum number of milliseconds to wait for a connection checkout.
+  // Any attempt to retrieve a connection from this pool that exceeds the set limit will throw an
+  // SQLException.
+  knex.client.pool.createTimeoutMillis = 30000; // 30 seconds
+  // 'idleTimeoutMillis' is the maximum amount of time a connection can sit in the pool. Connections that
+  // sit idle for this many milliseconds are retried if idleTimeoutMillis is exceeded.
+  knex.client.pool.idleTimeoutMillis = 600000; // 10 minutes
+  // [END cloud_sql_connection_timeout]
+  // [START cloud_sql_connection_backoff]
+  // 'createRetryIntervalMillis' is how long to idle after failed connection creation before trying again
+  knex.client.pool.createRetryIntervalMillis = 200; // 0.2 seconds
+  // [END cloud_sql_connection_backoff]
+  // [START cloud_sql_connection_lifetime]
+  // 'acquireTimeoutMillis' is the maximum possible lifetime of a connection in the pool. Connections that
+  // live longer than this many milliseconds will be closed and reestablished between uses. This
+  // value should be several minutes shorter than the database's timeout value to avoid unexpected
+  // terminations.
+  knex.client.pool.acquireTimeoutMillis = 600000; // 10 minutes
+  // [START cloud_sql_connection_lifetime]
+
+  return knex;
+  // [END cloud_sql_postgreSQL_connection_pool]
+}
+
+// [START cloud_sql_example_statement]
+/**
+ * Insert a vote record into the database.
+ *
+ * @param {object} knex The Knex connection object.
+ * @param {object} vote The vote record to insert.
+ * @returns {Promise}
+ */
+function insertVote(knex, vote) {
+  return knex('votes').insert(vote);
+}
+// [END cloud_sql_example_statement]
+
+/**
+ * Retrieve the latest 5 vote records from the database.
+ *
+ * @param {object} knex The Knex connection object.
+ * @returns {Promise}
+ */
+function getVotes(knex) {
+  return knex
+    .select('candidate', 'time_cast')
+    .from('votes')
+    .orderBy('time_cast', 'desc')
+    .limit(5)
+    .then(results => {
+      return results;
+    });
+}
+
+/**
+ * Retrieve the total count of records for a given candidate
+ * from the database.
+ *
+ * @param {object} knex The Knex connection object.
+ * @param {object} candidate The candidate for which to get the total vote count
+ * @returns {Promise}
+ */
+function getVoteCount(knex, candidate) {
+  return knex('votes')
+    .count('vote_id')
+    .where('candidate', candidate)
+    .then(result => {
+      return result;
+    });
+}
+
+app.get('/', (req, res) => {
+  var tabsTotalVotes;
+  var spacesTotalVotes;
+  // Query the total count of "TABS" from the database.
+  getVoteCount(knex, 'TABS').then(tabsResult => {
```

Comment:
Since you require Node 8, I'd much prefer seeing this with `async/await` instead of Promies.

Reply:
Done.